image/templates/helm/stackrox-central/internal/defaults.yaml.htpl

```diff
@@ -225,11 +225,11 @@ defaults:
                 operator: DoesNotExist
       resources:
         requests:
-          memory: "1500Mi"
-          cpu: "1000m"
+          memory: "500Mi"
+          cpu: "1500m"
         limits:
-          memory: "3Gi"
-          cpu: "2000m"
+          memory: "2000Mi"
```

Comment:
I'm wondering why we'd want to decrease the memory limit. If we were to one day need it, now we don't have the chance. Perhaps we may never need that much (I honestly can't recall why the 3Gi was chosen), so I just want to know more about your thoughts.

I'm thinking about where a lot of memory usage may come from for the Indexer. I'm wondering if it's just the Index Reports themselves

Reply:
> I'm wondering why we'd want to decrease the memory limit.

Thanks, that was a though crossing my mind and I am glad someone else thought of that as well.

There is no actual gain in setting a lower limit. In fact, I think the limit is more about limiting a memory-hungry Indexer from causing resource-starvation or noise to neighbours. The big benefit of leaving a higher limit is to avoid unexpected OOMs from legitimate spiks in memory consumption, specially from scenarios we haven't fully tested yet. So, I'm OK on keeping the original limit, if everybody is onboard.

Reply:
I think I'd prefer to keep the current limit at this time. I think the limit is a more nice-to-have, and the scheduler only considers the requests, anyway. I don't think I've heard complaints about the current limit level, so I'm ok with keeping it as-is at this time. I'm open to hearing counter points (I know our cloud services and OpenShift tend to prefer to not set any limits on resources)

Reply:
> I think I'd prefer to keep the current limit at this time. I think the limit is a more nice-to-have, and the scheduler only considers the requests, anyway.

I'm OK with that. But maybe "nice-to-have" is an oversimplification. It might be desirable to set a reasonable memory limit and allow the pod to be OOMKilled when it leaks, instead of silently causing churn in the node/cluster.

But yes, the 2Gi is just 2 times the observed memory consumption at 500 images, while the requests are coming from the base case of 100 images. So, 3Gi would probably be fine.

> I don't think I've heard complaints about the current limit level

I think actually we didn't hear anyone talking about limits because we never released an Indexer with a noticeable memory leak so far.